tests/components/RangeInput.test.tsx

```diff
@@ -0,0 +1,257 @@
+    userEvent.type(minTextbox, '10');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    userEvent.click(screen.getByText('Apply'));
+    expect(actions.setFilterOption).toHaveBeenCalledWith({
+      displayName: 'Over 10',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        start: {
+          matcher: '$ge',
+          value: 10
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing input and no state is set', async () => {
+    renderRangeInput(filterContextValue);
+    const actions = spyOnActions();
+    const minTextbox = screen.getAllByRole('textbox')[0];
+    userEvent.type(minTextbox, '10');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(minTextbox).toHaveValue('');
+    expect(actions.setFilterOption).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe('Renders correctly for max input', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+
+  it('renders input value, clear, and apply option when inputing max value', async () => {
+    renderRangeInput(filterContextValue);
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    expect(screen.getByText('Apply')).toBeDefined();
+  });
+
+  it('record proper values in state when applying max', async () => {
+    renderRangeInput(filterContextValue);
+    const actions = spyOnActions();
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    userEvent.click(screen.getByText('Apply'));
+    expect(actions.setFilterOption).toHaveBeenCalledWith({
+      displayName: 'Up to 20',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        end: {
+          matcher: '$le',
+          value: 20
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing input and no state is set', async () => {
+    renderRangeInput(filterContextValue);
+    const actions = spyOnActions();
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(maxTextbox).toHaveValue('');
+    expect(actions.setFilterOption).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe('Renders correctly for min and max inputs', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+
+  it('renders correctly when inputing range and applies proper values in state', async () => {
```

Comment:
Similar to the way you combined the "render correctly" and "apply to state" tests, lets update the two describe block above to do the same